datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestBroadcastStrategy.java

```diff
@@ -279,4 +280,21 @@ public void testRebalanceTasksWhenNewInstanceIsAdded() {
 
     Assert.assertEquals(newAssignment.get(instance4).size(), expectedNumTasksPerInstance);
   }
+
+  @Test
+  public void testDontCreateNewTasksWhenInstanceChanged() {
+    String[] instances = new String[]{"instance1", "instance2", "instance3"};
+    List<DatastreamGroup> datastreams = generateDatastreams("ds", 5);
+    BroadcastStrategy strategy = new BroadcastStrategy(Optional.empty());
+    Map<String, Set<DatastreamTask>> assignment =
+        strategy.assign(datastreams, Arrays.asList(instances), new HashMap<>());
+    instances = new String[]{"instance1", "instance2", "instance4"};
+
+    Map<String, Set<DatastreamTask>> newAssignment = strategy.assign(datastreams, Arrays.asList(instances), assignment);
+    Set<DatastreamTask> oldAssignmentTasks = assignment.values().stream().flatMap(Set::stream).collect(Collectors.toSet());
+    Set<DatastreamTask> newAssignmentTasks = newAssignment.values().stream().flatMap(Set::stream).collect(Collectors.toSet());
```

Comment:
nit: Names are slightly inconsistent; we get `oldAssignmentTasks` from `assignment` but `newAssignmentTasks` from `newAssignment`.

How about `firstAssignment` -> `firstAssignmentTasks` and `secondAssignment` -> `secondAssignmentTasks`?

Reply:
I kept the names inline with rest of the tests in the class for uniformity. Please let me know if you feel otherwise.

Reply:
Understood. No problem. Feel free to keep them as is.